packages/dds/tree/src/feature-libraries/contextuallyTyped.ts

```diff
@@ -334,6 +312,14 @@ function shallowCompatibilityTest(
 	}
 	// For now, consider all not explicitly typed objects shallow compatible.
 	// This will require explicit differentiation in polymorphic cases rather than automatic structural differentiation.
+
+	// Special case primitive schema to to not be compatible with data with local fields.
```

Comment:
This is the feature this PR started as. It makes contextual node data able to disambiguate nodes with fields from being primitives without an explicit type label.

Reply:
suggestion: "to to" -> "to"

Reply:
```suggestion
	// Special case primitive schema to not be compatible with data with local fields.
```